docs/core-design.md

```diff
@@ -0,0 +1,354 @@
+</p>
+
+### Positioning in DRA
+
+To better understand how Multi-Network will function with DRA we want to define 3 main use cases:
+
+1. Direct DRA device selection \- here are the AI/ML use cases, or simple device handling for which users do not need centralized configuration.  
+2. PodNetwork as Cluster-Level construct to enable orchestration of a common set of configurations \- this gives cluster administrators means to easily maintain uniformity. This design focuses on this area.  
+3. PodNetwork as a handle to enable Kubernetes networking primitives: Services, NetworkPolicies, etc.
+
+<p align="center">
+  <img src="images/image8.png?raw=true"/>
+</p>
+
+## Detailed design
+
+### PodNetwork CRD
+
+This design defines a PodNetwork object. This will be a CustomResourceDefinition (CRD). PodNetwork’s form is generic and vendor-neutral. This API object is what we wish to use for the future integration with the rest of the Kubernetes ecosystem.
+
+#### Motivation
+
+As of Today, there is no core object that represents the network that the Pod connects to. It represents the administrative resource at the cluster scope for configuration governance for the Pod connection, which provides the cluster administrators centralized control over what is used by the cluster workloads. Furthermore, this will allow workloads to select more than 1 such network configuration to attach to. In such situations, we want to be able to differentiate where specific Kubernetes functionality needs to be applied, e.g. serviceability which uses labels and namespace placement as the identifiers of what workloads to apply to, this object will be another indicator which interfaces of a Pod to use.
+
+##### Alternatives
+
+DRA DeviceClass \- one of the considerations was to re-use the DeviceClass object that comes with the DRA. The problem of reusing this object is that this is not strictly a network-specific object, DeviceClass can be used for any resource, e.g. a GPU. This will introduce confusion in integrating it with other Kubernetes functionality related to networking.
+
+#### API
+
+PodNetwork object is described as follows:
+
+```go
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+
+// PodNetwork represents a logical network in Kubernetes Cluster.
+type PodNetwork struct {
+        metav1.TypeMeta   `json:",inline"`
+        metav1.ObjectMeta `json:"metadata,omitempty"`
+
+        Spec   PodNetworkSpec   `json:"spec"`
+        Status PodNetworkStatus `json:"status,omitempty"`
+}
+
+// PodNetworkSpec contains the specifications for podNetwork object
+type PodNetworkSpec struct {
+
+        // Enabled is used to administratively enable/disable a PodNetwork.
+        // When set to false, PodNetwork Ready condition will be set to False.
+        // Defaults to True.
+        //
+        // +kubebuilder:default=true
+        Enabled bool `json:"enabled,"`
```

Comment:
so, if I toggle this to false it means all Pods in that network will be offline?